website/static/js/profile.js

```diff
@@ -33,11 +33,25 @@ var cleanByRule = function(rule) {
 
 var noop = function() {};
 
+
+
+
+
 var SerializeMixin = function() {};
 
 /** Serialize to a JS Object. */
 SerializeMixin.prototype.serialize = function() {
-    return ko.toJS(this);
+    var serializedData = ko.toJS(this);
+    
+    // Remove blank websites from profileWebsites array before save
+    if (serializedData.urls.crud = "/api/v1/settings/social/") {
+        var profileWebsites = serializedData.trackedProperties[0];
+        if (profileWebsites.length > 1) {
+            serializedData.profileWebsites = $.grep(profileWebsites,function(n){ return(n) });
```

Comment:
Is this for filtering inputs that are truthy? If so, you should be able to use `Array.filter`.